pkgs/development/python-modules/hatchling/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+
+# All runtime dependencies
+, editables
+, importlib-metadata
+, packaging
+, pathspec
+, pluggy
+, tomli
+}:
+
+buildPythonPackage rec {
+  pname = "hatchling";
+  version = "0.20.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1f961cedeed8535e74ee5c6ea6615c74bbd21363f3e294b18ac33e5cf44bb664";
+  };
+
+  propagatedBuildInputs = [
+    editables
+    packaging
+    pathspec
+    pluggy
+    tomli
+  ] ++ lib.optionals (pythonOlder "3.8") [
+    importlib-metadata
+  ];
+
+  pythonImportsCheck = [ "hatchling.build" ];
+
+  doCheck = false;
```

Comment:
Why? The project has tests https://github.com/ofek/hatch/tree/master/tests and missing pythonImportsCheck.

Reply:
Because the tests try to pull library from the internet.

```
running install tests
<<<<< Copying backend >>>>>
<<<<< Building backend >>>>>
* Creating virtualenv isolated environment...
* Getting dependencies for wheel...
* Installing packages in isolated environment... (editables>=0.2; python_version > "3", importlib-metadata; python_version < "3.8", packaging>=21.3; python_version > "3", packaging~=20.9; python_version < "3", pathspec>=0.9, pluggy>=1.0.0; python_version > "3", pluggy~=0.13; python_version < "3", tomli>=1.2.2; python_version > "3", toml~=0.10.2; python_version < "3")
Ignoring packaging: markers 'python_version < "3"' don't match your environment
Ignoring toml: markers 'python_version < "3"' don't match your environment
Ignoring pluggy: markers 'python_version < "3"' don't match your environment
Ignoring importlib-metadata: markers 'python_version < "3.8"' don't match your environment
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff59be410>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff59bfe80>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff59be0e0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff59be110>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff56fa110>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
ERROR: Could not find a version that satisfies the requirement pathspec>=0.9 (from versions: none)
ERROR: No matching distribution found for pathspec>=0.9
```

Reply:
Ah, it tries to install things into a virtualenv.